File: programming/lab2/src/main/java/moves/status/Refresh.java
package moves.status;

import ru.ifmo.se.pokemon.*;

/**
 * Cleanses the user of a burn, paralysis, or poison.
 */
public class Refresh extends StatusMove {
    public Refresh() {
        super(Type.NORMAL, 0, 100);
    }

    @Override
    protected boolean checkAccuracy(Pokemon pokemon, Pokemon pokemon1) {
        return true;
    }

    @Override
    protected String describe() {
        return "использует Refresh";
    }

    @Override
    protected void applySelfEffects(Pokemon p) {
        if (p.getCondition() == Status.BURN || p.getCondition() == Status.PARALYZE || p.getCondition() == Status.POISON) {
            Effect normalCondition = new Effect();
            normalCondition.condition(Status.NORMAL);
            p.setCondition(normalCondition);
        }
    }
}

File: programming/lab2/app/src/main/kotlin/org/itmo/moves/status/Refresh.kt
package org.itmo.moves.status

import ru.ifmo.se.pokemon.*

class Refresh : StatusMove(Type.NORMAL, 0.0, 100.0) {
    override fun applySelfEffects(p0: Pokemon) {
        if (p0.condition in arrayOf(Status.PARALYZE, Status.POISON, Status.BURN)) {
            p0.setCondition(Effect().condition(Status.NORMAL))
        }
    }

    override fun checkAccuracy(p0: Pokemon?, p1: Pokemon?): Boolean {
        return true
    }

    override fun describe(): String {
        return "использует Refresh"
    }
}